src/app/components/app-node-utilizations/app-node-utilizations.component.ts

```diff
@@ -0,0 +1,150 @@
+import { SchedulerService } from '@app/services/scheduler/scheduler.service';
+
+
+@Component({
+  selector: 'app-node-utilizations',
+  templateUrl: './app-node-utilizations.component.html',
+  styleUrls: ['./app-node-utilizations.component.scss']
+})
+export class AppNodeUtilizationsComponent implements OnInit, OnChanges {
+  nodeUtilizations: NodeUtilization[] = [];
+
+  // input data for vertical bar chart, key is resource type
+  bucketList: string[] = [];                                          // one bucket list for all resource types, length should be exactly 10
+  barChartDataSets: BarChartDataSet[] = new Array<BarChartDataSet>(); // one dataset for each type
+
+  @Input() partitionSelected: string = "";
+
+  constructor(
+    private scheduler: SchedulerService
+  ) { }
+
+  ngOnInit() {
+    this.reloadBarChartData()
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (
+      changes['partitionSelected']
+    ) {
+      this.reloadBarChartData()
+    }
+  }
+
+  reloadBarChartData() {
+    this.scheduler.fetchNodeUtilizationsInfo().subscribe((data) => {
+      let nodeUtilizationsInfo: NodeUtilizationsInfo[] = data
+      for (let i = 0; i < nodeUtilizationsInfo.length; i++) {
+        if (nodeUtilizationsInfo[i].partition === this.partitionSelected) {
+          let nodeUtilizations = nodeUtilizationsInfo[i].utilizations
+          this.fetchBarChartData(nodeUtilizations)
+          break;
+        }
+      }
+    });
+  }
+
+  fetchBarChartData(nodeUtilizations: NodeUtilization[]) {
+    let barChartDataSets = new Array<BarChartDataSet>();
+    if (nodeUtilizations.length === 0) {
+      // clean bar chart data
+      this.barChartDataSets = barChartDataSets;
+      return;
+    }
+
+    let colorMapping = this.generateColorMapping(
+      nodeUtilizations.map((nodeUtilization) => (nodeUtilization.type))
+    );
+
+    for (let i = 0; i < nodeUtilizations.length; i++) {
+      let type = nodeUtilizations[i].type;
+      let utilization = nodeUtilizations[i].utilization
+      let borderWidth = 1
+
+      if (i === 0) {
+        // get bucketList only from the first type of node utilization
+        // should always be 10 buckets. (ranging from 0% to 100%).
+        this.bucketList = utilization.map((item) => item.bucketName);
+      }
+      let bucketValues = utilization.map((item) => item.numOfNodes);
+      barChartDataSets.push(new BarChartDataSet(
+        type,
+        bucketValues,
+        this.calculateAvgUtilization(bucketValues),
+        colorMapping.get(type) ?? DEFAULT_BAR_COLOR,
+        borderWidth,
+        utilization.map((item) => this.getBarDescription(item.nodeNames))
+      ))
+    }
+
+    // sort by resource type first, then sort by avg utilization rate
+    barChartDataSets.sort((a, b) => CommonUtil.resourcesCompareFn(a.label, b.label));
+    barChartDataSets.sort((a, b) => b.avgUtilizationRate - a.avgUtilizationRate);
+    barChartDataSets = barChartDataSets.slice(0, 5); // only show top 5 resources
```

Comment:
Since the legend uses checkboxes, freeing up extra space for resource types, is it still necessary to limit the display to only the top 5 resources?

Reply:
Hi @wusamzong, Agree to show more resourced because we allow users to select what they want to see,  I think change the limit to 10 is acceptible in the current layout.

Reply:
Changed the maximum resource type number to 10.

(6 types in my environement)
<img width="386" alt="image" src="https://github.com/apache/yunikorn-web/assets/26764036/5d6159fa-d054-4586-9d26-cb06d0998c4d">